compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1664,12 +1665,20 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
                         suggests_in_band = true;
                         (generics.span, format!("<{}>", lifetime_ref))
                     };
-                    err.span_suggestion(
-                        span,
-                        &format!("consider introducing lifetime `{}` here", lifetime_ref),
-                        sugg,
-                        Applicability::MaybeIncorrect,
-                    );
+                    if !span.from_expansion() {
+                        err.span_suggestion(
+                            span,
+                            &format!("consider introducing lifetime `{}` here", lifetime_ref),
+                            sugg,
+                            Applicability::MaybeIncorrect,
+                        );
+                    } else if suggest_note {
+                        suggest_note = false; // Avoid displaying the same help multiple times.
+                        err.help(&format!(
+                            "consider introducing lifetime `{}` to the item's generics",
+                            lifetime_ref,
```

Comment:
can you add a note here mentioning that we're inside of an expansion?

I think otherwise these two nearly identical error messages in the added test case might be confusing

Reply:
What do you think about rewriting this to "lifetime `{}` is missing in an item created through a proc macro"?

Reply:
Check out the new output